src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs

```diff
@@ -140,16 +140,22 @@ internal static FunctionSignatureWithReturnType MatchSignatureToBuiltInFunction(
 
         /// <summary>
         /// Finds all signatures for the given function name.
+        /// Search in BuiltIn uri functions first, and then in custom uri functions.
         /// </summary>
         /// <param name="functionName">The function to get the signatures for.</param>
         /// <returns>The signatures which match the supplied function name.</returns>
-        internal static FunctionSignatureWithReturnType[] GetBuiltInFunctionSignatures(string functionName)
+        internal static FunctionSignatureWithReturnType[] GetUriFunctionSignatures(string functionName)
         {
-            // Try to find the function in our built-in functions
             FunctionSignatureWithReturnType[] signatures;
-            if (!BuiltInFunctions.TryGetBuiltInFunction(functionName, out signatures))
+
+            // Try to find the function in the user custom functions
+            if (!CustomUriFunctions.TryGetCustomFunction(functionName, out signatures))

```

Comment:
@YogiBear52, What i get from the code is that the AddCustomUriFunction could add a new custom function which has a different signature from all the same named built-in functions. if the signatures are same, the code will throw CustomUriFunctions_AddCustomUriFunction_BuiltInExistsFullSignature exception, right?
So i think, maybe we should return both the built-in and custom functions with the name if 'overrideBuiltInFunction' is true.